lua/rip-substitute/matches.lua

```diff
@@ -0,0 +1,221 @@
+local M = {}
+local rg = require("rip-substitute.rg-operations")
+local utils = require("rip-substitute.utils")
+
+---@return string
+---@return string
+local function getSearchAndReplaceValuesFromPopup()
+	local config = require("rip-substitute.config").config
+	local state = require("rip-substitute.state").state
+
+	local toSearch, toReplace = unpack(vim.api.nvim_buf_get_lines(state.popupBufNr, 0, -1, false))
+	if config.regexOptions.autoBraceSimpleCaptureGroups then
+		toReplace = toReplace:gsub("%$(%d+)", "${%1}")
+	end
+	return toSearch, toReplace
+end
+
+---@class RipSubstituteMatch
+---@field row number
+---@field col number
+---@field matchedText string
+---@field replacementText string
+
+---@return RipSubstituteMatch[]|nil,string|nil
+function M.getMatches()
+	local toSearch, toReplace = getSearchAndReplaceValuesFromPopup()
+	if not toSearch then return end
+	local code, matched = rg.runRipgrep {
+		"--line-number",
+		"--column",
+		"--vimgrep",
+		"--no-filename",
+		"--only-matching",
+		toSearch,
+	}
+	if code ~= 0 then return nil, "[1]could not get rg matches" end
+	if #matched == 0 then return {}, nil end
+
+	---@type RipSubstituteMatch[]
+	local matches = utils.map(
+		matched,
+		---@param line string
+		---@param i integer
+		function(line, i)
+			local rowStr, colStr, text = line:match("^(%d+):(%d+):(.*)")
+			---@type RipSubstituteMatch
+			local match = {
+				row = tonumber(rowStr) - 1,
+				col = tonumber(colStr) - 1,
+				matchedText = text,
+				replacementText = "",
+			}
+			return match
+		end
+	)
+
+
+	if toReplace and toReplace ~= "" then
+		for i, match in ipairs(matches) do
+			local rgArgs = {
+				"--line-number",
+				"--column",
+				"--vimgrep",
```

Comment:
I think `--vimgrep` is just a shorthand that bundles a bunch of other options. I think it actually not needed, since the other options already cover it

Reply:
removed 👍